### PR TITLE
[BB-6458] Fix tasks that create django users for IDAs

### PR DESCRIPTION
**Pull request by Cup0fCoffee**
### Description

Tasks that create django users for IDAs would run when creating django users for `edx-platform`. In their `manage.py` command they would omit `lms` or `cms` (because they are targeting IDAs), and that would cause the tasks to fail, because when running `manage.py` commands in `edx-platform`, passing `lms` or `cms` is mandatory.

Since these tasks should only run when targeting IDAs, added a predicate, that checks that `service` variable is not an empty string. That way these tasks will only be executed when the playbook is used to create a django user for an IDA.

### Links

[BB-6458](https://tasks.opencraft.com/browse/BB-6458)

### CODE REVIEW

File: playbooks/manage_edxapp_users_and_groups.yml [hunk 1]
```diff
@@ -197,7 +197,7 @@
         {% if item.get('permissions', []) | length %}--permissions {{ item.permissions | default([]) | map('quote') | join(' ') }}{% endif %}
         {% if item.get('remove') %}--remove{% endif %}
       with_items: "{{ django_groups }}"
-      when: (not group_environment) or group_environment in item.environments
+      when: (service | bool) and (not group_environment) or group_environment in item.environments
```
Comment (kaustavb12) [inline playbooks/manage_edxapp_users_and_groups.yml]: According to multiple sources that I found ([source1](https://stackoverflow.com/questions/36912726/ansible-how-to-test-that-a-registered-variable-is-not-empty), [source2](https://github.com/ansible/ansible-lint/issues/457#issuecomment-455038475)), the expression `var | bool` would evaluate for `False`, even if var is a non-empty string.

The most robust method which is recommended is 
```suggestion
      when: (service | length > 0) and (not group_environment) or group_environment in item.environments
```

File: playbooks/manage_edxapp_users_and_groups.yml [hunk 2]
```diff
@@ -214,7 +214,7 @@
         {% if item.get('unusable_password') %}--unusable-password{% endif %}
         {% if item.get('initial_password_hash') %}--initial-password-hash {{ item.initial_password_hash | quote }}{% endif %}
       with_items: "{{ django_users }}"
-      when: not item.get('unusable_password')
+      when: (service | bool) and not item.get('unusable_password')
```
Comment (kaustavb12) [inline playbooks/manage_edxapp_users_and_groups.yml]: Same as above

File: playbooks/manage_edxapp_users_and_groups.yml [hunk 3]
```diff
@@ -235,7 +235,7 @@
         {% if item.get('unusable_password') %}--unusable-password{% endif %}
         {% if item.get('initial_password_hash') %}--initial-password-hash {{ item.initial_password_hash | quote }}{% endif %}
       with_items: "{{ django_users }}"
-      when: item.get('unusable_password')
+      when: (service | bool) and item.get('unusable_password')
```
Comment (kaustavb12) [inline playbooks/manage_edxapp_users_and_groups.yml]: Same as above

**Review by kaustavb12 — APPROVED** [review]
:+1:

Approved, subject to the resolution of the review comments.

- [X] I tested this: Verified [test instance](https://manage.opencraft.com/instance/36200/) is successfully provisioned with this branch and lms users have been populated
- [X] I read through the code

File: playbooks/manage_edxapp_users_and_groups.yml (hunk 1 above)
Comment (Cup0fCoffee) [reply playbooks/manage_edxapp_users_and_groups.yml]: @kaustavb12 Good catch! Changed.

**Pull request merged**